### ISSUE TITLE
perft allows castling into check in chess960 (extra move / segfault)

### ISSUE THREAD

**Issue by theoden8** [official-stockfish/Stockfish#3605]
Hey,

The case when a king castles into check in chess960 causes seg-fault:

FEN: [4k3/8/8/8/8/8/8/rR2K1N1 w Q - 0 1](https://lichess.org/analysis/standard/4k3/8/8/8/8/8/8/rR2K1N1_w_Q_-_0_1)

I found this case from [here](https://chess.stackexchange.com/questions/27685/official-rules-for-castling-in-chess960-is-there-a-loophole).

For me the fix was to check if the rook is not pinned.

Best,
Kirill

**Comment by BM123499** [comment]
Hi, @theoden8.
We indeed verify if the rook is pinned, I just tested this position on my machine and it was okay.
I believe you forgot to activate chess960 handling on your test, by entering `setoption name UCI_Chess960 value true` before setting the position.
We only verify for those unique chess960 positions if the option `UCI_Chess960` is true beforehand.
The default value for `UCI_Chess960` is false.
